src/de.rs

```diff
@@ -172,9 +172,10 @@ impl<'a, 'de> de::Deserializer<'de> for &'a mut Deserializer<'de> {
         V: Visitor<'de>,
     {
         match *self.input {
+            Value::Null => visitor.visit_none(),
```

Comment:
I'm not sure either, I tossed it in because it fixed a problem I was having deserializing an Option. If I can reproduce this, it deserves a unit test  and to go into its own PR